src/Storages/StorageMerge.cpp

```diff
@@ -583,7 +583,7 @@ std::vector<ReadFromMerge::ChildPlan> ReadFromMerge::createChildrenPlans(SelectQ
             database_name,
             table_name,
             RowPolicyFilterType::SELECT_FILTER);
-        if (row_policy_filter_ptr)
+        if (row_policy_filter_ptr && row_policy_filter_ptr->expression)
```

Comment:
Can you add a comment somewhere about why the `USING 1` statement results in `expression = nullptr`?

Reply:
Hello @pufit , as far as I can see zero expression is a normal course of actions.

The obvious reason is optimization in FiltersMixer::getResult
```
                if (tryGetLiteralBool(result.get(), value) && value)
                    result = nullptr; /// The condition is always true, no need to check it.
```
May be it the the only one, not sure.

It seems that instead of checking "row_policy_filter_ptr->expression" I should call RowPolicyFilter::empty()
which probably makes a comment you are requesting redundant.

Reply:
We have exactly the same
```
            if (row_policy_filter && !row_policy_filter->empty())
```
in InterpreterSelectQuery ctor, so this check seems a common pattern.